diff -p -up imlib-1.9.15/configure.in.no-locincpth imlib-1.9.15/configure.in
--- imlib-1.9.15/configure.in.no-locincpth	2004-09-23 03:13:45.000000000 +0200
+++ imlib-1.9.15/configure.in	2009-08-16 17:03:49.000000000 +0200
@@ -141,7 +141,11 @@ else
 fi
 AC_SUBST(GX_LIBS)
 
-CPPFLAGS="$CPPFLAGS -I$includedir -I$prefix/include"
+dnl Get rid of system include dirs
+CPPFLAGS=`echo "$CPPFLAGS " | sed -e "s|-I/usr/include ||g"`
+if [[ `eval $includedir` != "/usr/include" ]]; then
+  CPPFLAGS="$CPPFLAGS -I$includedir"
+fi
 
 SUPPORT_LIBS=""
 
diff -p -up imlib-1.9.15/imlib-config.in.no-locincpth imlib-1.9.15/imlib-config.in
--- imlib-1.9.15/imlib-config.in.no-locincpth	2004-08-27 18:03:11.000000000 +0200
+++ imlib-1.9.15/imlib-config.in	2009-08-16 17:03:49.000000000 +0200
@@ -59,11 +59,11 @@ while test $# -gt 0; do
       ;;
     --libs)
       libdirs=-L@libdir@
-      echo $libdirs -lImlib @LDFLAGS@ $gx_libs @X_LIBS@ @X_PRE_LIBS@ -lXext -lX11 @X_EXTRA_LIBS@
+      echo -lImlib @LDFLAGS@ $gx_libs @X_LIBS@ @X_PRE_LIBS@ -lXext -lX11 @X_EXTRA_LIBS@
       ;;
     --libs-gdk)
       libdirs=-L@libdir@
-      echo $libdirs -lgdk_imlib$gdk_gx_libs `@GTK_CONFIG@ --libs` $gdk_libs
+      echo -lgdk_imlib$gdk_gx_libs `@GTK_CONFIG@ --libs` $gdk_libs
       ;;
     *)
       echo "${usage}" 1>&2
